shared/src/main/scala/async/futures.scala

```diff
@@ -362,8 +361,11 @@ enum TaskSchedule:
   */
 class Task[+T](val body: (Async, AsyncOperations) ?=> T):
 
+  /** Run the current task and returns the result. */
+  def run()(using Async, AsyncOperations): T = body
+
   /** Start a future computed from the `body` of this task */
-  def run(using Async, AsyncOperations) = Future(body)
+  def start()(using Async.Spawn, AsyncOperations) = Future(body)
```

Comment:
```suggestion
  def start()(using async: Async, spawnable: Async.Spawn & async.type, AsyncOperations) = Future(body)
```
Here, too, we want to capture only innermost `Async` instances (and require them to be `Spawn)`

Reply:
Makes sense!